src/Provider/AbstractProvider.php

```diff
@@ -442,36 +471,25 @@ public function getAccessToken($grant, array $options = [])
 
 
     /**
-     * Get a request instance.
+     * Get a request instance, optionally authenticated using an access token.
      *
      * Creates a PSR-7 compatible request instance that can be modified.
-     * The request is not automatically authenticated.
      *
      * @param  string $method
      * @param  string $url
+     * @param  AccessToken|string $token
      * @param  array  $options Any of "headers", "body", and "protocolVersion".
      * @return RequestInterface
      */
-    public function getRequest($method, $url, array $options = [])
+    public function getRequest($method, $url, $token = null, array $options = [])
```

Comment:
I thought about this pretty hard before I went ahead with it. When would a provider (or API wrapper) want to make an request that is not authenticated? The only time that that make sense is for the request to authenticate, which is internal.

> having a hard $token requirement for authenticated requests allows for more accurate hinting

Didn't you argue against strong access token hinting earlier? Should be able to get a request using a string-value token, so I don't think accurate hinting is an argument here.